test/common/signal/signals_test.cc

```diff
@@ -205,4 +220,71 @@ TEST(FatalErrorHandler, DontAllocateMemory) {
   EXPECT_MEMORY_EQ(allocated_after_call, allocated_before_call);
 }
 
+TEST(FatalErrorHandler, ShouldOnlyBeAbleToRegisterFatalActionsOnce) {
+  EXPECT_DEATH(
+      {
+        auto safe_actions =
+            std::make_unique<std::list<const Server::Configuration::FatalActionPtr>>();
+        auto unsafe_actions =
+            std::make_unique<std::list<const Server::Configuration::FatalActionPtr>>();
+        FatalErrorHandler::registerFatalActions(std::move(safe_actions), std::move(unsafe_actions),
+                                                nullptr);
```

Comment:
Can this be done outside the EXPECT_DEATH?

Reply:
The issue is that we're dealing with static memory.

I'll set up a test fixture that will have a `SetUpTestSuite()` class that will register the statics once for these tests.

Reply:
`SetupTestSuite()` and static memory with mocks was a bit bad -- ended up putting a "reset module" function for test use only in `fatal_error_handler.cc` and narrowed interfaces to prevent the need for mocks.